modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCachePartitionExchangeManager.java

```diff
@@ -1856,14 +1860,17 @@ public void dumpDebugInfo(@Nullable GridDhtPartitionsExchangeFuture exchFut) thr
         exchWorker.dumpExchangeDebugInfo();
 
         if (!readyFuts.isEmpty()) {
-            U.warn(diagnosticLog, "First 5 pending affinity ready futures [total=" + readyFuts.size() + ']');
+            int warningsLimit = IgniteSystemProperties.getInteger(IGNITE_DIAGNOSTIC_WARN_LIMIT, 5);
+
+            U.warn(diagnosticLog, "First " + warningsLimit + " pending affinity ready futures [total=" +
+                readyFuts.size() + ']');
 
             int cnt = 0;
 
             for (AffinityReadyFuture fut : readyFuts.values()) {
                 U.warn(diagnosticLog, ">>> " + fut);
 
-                if (++cnt == 5)
+                if (++cnt == warningsLimit)
```

Comment:
It looks like we can't have IGNITE_DIAGNOSTIC_WARN_LIMIT = 0 since at least one message will always be printed. Can we move this check up (while avoiding off-by-one error) so that we could completely disable those messages with IGNITE_DIAGNOSTIC_WARN_LIMIT=0? Here and below, everywhere where it is checked.

Reply:
That's a good point. Fixed.